specification/storage/resource-manager/Microsoft.Storage/2017-06-01/storage.json

```diff
@@ -757,6 +816,47 @@
             "name": "SkuTier",
             "modelAsString": false
           }
+        },
+        "resourceType":{
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of the resource, usually it is 'storageAccounts'."
+        },
+        "kind": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Indicates the type of storage account.",
+          "enum": [
+            "Storage",
+            "BlobStorage"
+          ],
+          "x-ms-enum": {
+            "name": "Kind",
+            "modelAsString": false
+          }
+        },
+        "locations": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The location of the resource. This will be one of the supported and registered Azure Geo Regions (e.g. West US, East US, Southeast Asia, etc.)."
+        },
+        "capabilities": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SKUCapability"
+          },
+          "description": "The capability information in the specified sku, including file encryption, network acls, change notification, etc."
+        },
+        "restrictions": {
```

Comment:
considered a breaking change - @ravbhatnagar  please review

Reply:
@JasonYang-MSFT - ChecknameAvailability API as defined in the ARM RPC should be used to indicate whether a given resource name is available or not. It seems like in this case you are returning more information that what is required in the contract. Why is it needed to return all this extra information like kind, resource type, locations etc. resource type goes in the request body. Locations go in the URL to verify local uniqueness. Not sure why you need kind or SKUCapability to be returned for this API.

Reply:
@ravbhatnagar Sorry, am I miss something here? 
This PR is about sku list operation, "ChecknameAvailability" is not updated in this PR and it is already in the swagger.

Reply:
Whoops.. @JasonYang-MSFT  - please ignore my earlier comment. The way the changes were diff'd in Github gave the impression that all these properties are added to the response of checkNameAvailability :). 
The only feedback here is that restrictions should have type, values and reasonCode properties and all these are require for restrictions. restriction.type is most likely locations today. And so the restriction.values will contain the list of locations where the skus is restricted. and then there will be a reason for the restriction in reasoncode. Does that make sense? and yes, this is not a breaking change.

Reply:
Yes, your comment is addressed. Please kindly take a look.